feat: add current_date, current_timestamp and current_timezone variables (#945)
substrait-io/substrait: site/docs/expressions/specialized_record_expressions.md

For single value expressions, these are a compact equivalent of `expression = value1 OR expression = value2 OR .. OR expression = valueN`. When using an expression of this type, two things are required; the types of the test expression and all value expressions that are related must be of the same type. Additionally, a function signature for equality must be available for the expression type used.

## Session Variables

Session variables are a special class of expressions whose behavior depends on the current session context. Substrait defines a set of common session variables and allows to define extension session variables.

| Session Variable | Description | Return Type |
| --------------------- | ----------- | ----------- |
| current_date | a non-deterministic, session dependent variable containing the current date | date |

> | current_timestamp | a non-deterministic, session dependent variable containing the current timestamp in current_timezone | PRECISION_TIMESTAMP_TZ<P> |

Reviewer: do we have a function defined to cast PRECISION_TIMESTAMP_TZ to PRECISION_TIMESTAMP? or do we want one returning PRECISION_TIMESTAMP as well?

For single value expressions, these are a compact equivalent of `expression = value1 OR expression = value2 OR .. OR expression = valueN`. When using an expression of this type, two things are required; the types of the test expression and all value expressions that are related must be of the same type. Additionally, a function signature for equality must be available for the expression type used.

## Session Variables

> Session variables are a special class of expressions whose behavior depends on the current session context. Substrait defines a set of common session variables and allows to define extension session variables.

Reviewer 2: ```suggestion
Session variables are a special class of expressions whose behavior depends on the current session context. Substrait defines a set of common session variables, but plan producers can also define extension session variables.
```
